File: ActivityDemo/app/src/main/java/com/example/a14512/activitydemo/utils/LogUtil.java
package com.example.a14512.activitydemo.utils;

import android.util.Log;

/**
 * @author 14512 on 2018/7/30
 */
public class LogUtil {

    public static void e(String tag, String msg) {
        Log.e(tag, msg);
    }

    public static void e(String msg) {
        e(getClassName(), msg);
    }

    public static void w(String tag, String msg) {
        Log.w(tag, msg);
    }

    public static void w(String msg) {
        w(getClassName(), msg);
    }

    public static void i(String tag, String msg) {
        Log.i(tag, msg);
    }

    public static void i(String msg) {
        i(getClassName(), msg);
    }

    public static void d(String tag, String msg) {
        Log.d(tag, msg);
    }

    public static void d(String msg) {
        d("LogUtil  "+getClassName(), msg);
    }

    public static void v(String tag, String msg) {
        Log.v(tag, msg);
    }

    public static void v(String msg) {
        v(getClassName(), msg);
    }

    /**
     * @return 当前的类名(simpleName)
     */
    public static String getClassName() {
        String result;
//        StackTraceElement thisMethodStack = Thread.currentThread().getStackTrace()[2];
        StackTraceElement thisMethodStack = (new Exception()).getStackTrace()[2];
        result = thisMethodStack.getClassName();
        int lastIndex = result.lastIndexOf(".");
        result = result.substring(lastIndex + 1);

        // TODO: 16/5/10 调试下 内部类问题
        // 剔除匿名内部类名
        int i = result.indexOf("$");
        return i == -1 ? result : result.substring(0, i);
    }
}

File: GithubDemo/app/src/main/java/com/example/githubdemo/common/utils/LogUtil.kt
package com.example.githubdemo.common.utils

import android.util.Log

object LogUtil {
    val ASSERT = 7
    val DEBUG = 3
    val ERROR = 6
    val INFO = 4
    val VERBOSE = 2
    val WARN = 5

    fun v(tag: String, msg: String) {
        Log.v(tag, msg)
    }

    fun v(tag: String, msg: String, tr: Throwable) {
        Log.v(tag, msg, tr)
    }

    fun d(tag: String, msg: String) {
        Log.d(tag, msg)
    }

    fun d(tag: String, msg: String, tr: Throwable) {
        Log.d(tag, msg, tr)
    }

    fun i(tag: String, msg: String) {
        Log.i(tag, msg)
    }

    fun i(tag: String, msg: String, tr: Throwable) {
        Log.i(tag, msg, tr)
    }

    fun w(tag: String, msg: String) {
        Log.w(tag, msg)
    }

    fun w(tag: String, msg: String, tr: Throwable?) {
        Log.i(tag, msg, tr)
    }

    fun e(tag: String, msg: String) {
        Log.e(tag, msg)
    }

    fun e(tag: String, msg: String, tr: Throwable?){
        Log.e(tag, msg, tr)
    }

    fun v(msg: String) {
        v(getClassName(), msg)
    }

    fun d(msg: String) {
        d(getClassName(), msg)
    }

    fun i(msg: String) {
        i(getClassName(), msg)
    }

    fun w(msg: String) {
        w(getClassName(), msg)
    }

    fun e(msg: String) {
        e(getClassName(), msg)
    }

    /**
     * @return 当前的类名(simpleName, 此时运行的类名)
     */
    private fun getClassName(): String {
        var result: String
        val thisMethodStack = Thread.currentThread().stackTrace[2]
        result = thisMethodStack.className
        val lastIndex = result.lastIndexOf(".")
        result = result.substring(lastIndex + 1)

        val i = result.indexOf("$")// 剔除匿名内部类名
        return if (i == -1) result else result.substring(0, i)
    }
}